client/my-sites/earn/memberships/products.tsx

```diff
@@ -23,36 +23,44 @@ import getFeaturesBySiteId from 'calypso/state/selectors/get-site-features';
 import siteHasFeature from 'calypso/state/selectors/site-has-feature';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
 import RecurringPaymentsPlanAddEditModal from '../components/add-edit-plan-modal';
+import { Product } from '../types';
 import { ADD_NEW_PAYMENT_PLAN_HASH, ADD_NEWSLETTER_PAYMENT_PLAN_HASH } from './constants';
 import RecurringPaymentsPlanDeleteModal from './delete-plan-modal';
 import MembershipsSection from './';
 import './style.scss';
 
+type MembersProductsSectionProps = {
+	section: string;
+	query?: {
+		[ key: string ]: string;
+	};
+};
+
 const showAddEditDialogInitially =
 	window.location.hash === ADD_NEW_PAYMENT_PLAN_HASH ||
 	window.location.hash === ADD_NEWSLETTER_PAYMENT_PLAN_HASH;
 
-function MembershipsProductsSection( { section, query } ) {
+function MembershipsProductsSection( { section, query }: MembersProductsSectionProps ) {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
 	const [ showAddEditDialog, setShowAddEditDialog ] = useState( showAddEditDialogInitially );
 	const [ showDeleteDialog, setShowDeleteDialog ] = useState( false );
-	const [ product, setProduct ] = useState( null );
+	const [ product, setProduct ] = useState< Product | null >( null );
 	const site = useSelector( ( state ) => getSelectedSite( state ) );
 	const features = useSelector( ( state ) => getFeaturesBySiteId( state, site?.ID ) );
 	const hasLoadedFeatures = features?.active.length > 0;
-	const products = useSelector( ( state ) => getProductsForSiteId( state, site?.ID ) );
+	const products: Product[] = useSelector( ( state ) => getProductsForSiteId( state, site?.ID ) );
```

Comment:
In future we could change this to use tan-query hook and get `isLoading` for free.